services/wireguard/connection/connection.go

```diff
@@ -127,38 +137,99 @@ func (c *Connection) Start(options connection.ConnectOptions) (err error) {
 	return nil
 }
 
+func (c *Connection) startConn(conf wg.StartConsumerConfig) (wg.ConnectionEndpoint, error) {
+	// We do not need port mapping for consumer, since it initiates the session
+	fakePortMapper := func(port int) (releasePortMapping func()) {
+		return func() {}
+	}
+
+	resourceAllocator := connectionResourceAllocator()
+	conn, err := endpoint.NewConnectionEndpoint(nil, resourceAllocator, fakePortMapper, 0)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create new connection endpoint")
+	}
+
+	log.Info().Msg("Starting connection endpoint")
+	if err := conn.Start(wg.StartConfig{
+		Consumer: &conf,
+	}); err != nil {
+		return nil, errors.Wrap(err, "failed to start connection endpoint")
+	}
+
+	return conn, nil
+}
+
+func (c *Connection) addProviderPeer(conn wg.ConnectionEndpoint, endpoint net.UDPAddr, publicKey string, natPunchingEnabled bool) error {
+	var keepAlivePeriodSeconds int
+	if natPunchingEnabled {
+		keepAlivePeriodSeconds = 25
```

Comment:
Why can't we have the same keep alive period for both with NAT punching and without it?